### PR TITLE
Minify the resources content at warmup

### PR DESCRIPTION
**Pull request by engahmeds3ed**
## Description

Minify resources' content at warmup as this also will solve some issues like sending all imports to the top of the file before calling treeshaker.

### CODE REVIEW

File: inc/Engine/Optimization/RUCSS/Controller/UsedCSS.php [hunk 1]
```diff
@@ -313,6 +314,11 @@ private function save_or_update_used_css( array $data ) {
 		$data['css'] = preg_replace( '/content\s*:\s*\'\\\\f/i', 'shaker-parser:\'dashf', $data['css'] );
 		$data['css'] = preg_replace( '/content\s*:\s*\'\\\\e/i', 'shaker-parser:\'dashe', $data['css'] );
 
+		$minifier = new MinifyCSS();
+		$minifier->add( $data['css'] );
```
Comment (remyperona) [inline inc/Engine/Optimization/RUCSS/Controller/UsedCSS.php]:
```suggestion
		$minifier = new MinifyCSS( $data['css'] );
```

File: inc/Engine/Optimization/RUCSS/Warmup/ResourceFetcher.php [hunk 2]
```diff
@@ -159,6 +160,22 @@ private function find_resources( string $html, string $type ) {
...
+	 *
+	 * @param string $path Path of the CSS file.
+	 * @param string $contents Contents of the CSS file.
+	 *
+	 * @return string
+	 */
+	private function prepare_css_content( string $path, string $contents ) : string {
+		$contents = trim( $this->rewrite_paths( $path, $path, $contents ) );
+		$minifier = new MinifyCSS();
+		$minifier->add( $contents );
```
Comment (remyperona) [inline inc/Engine/Optimization/RUCSS/Warmup/ResourceFetcher.php]:
```suggestion
		$minifier = new MinifyCSS( $contents );
```

**Review by remyperona — CHANGES_REQUESTED** [review]
Small suggestions

File: inc/Engine/Optimization/CSSTrait.php [hunk 3]
```diff
@@ -539,4 +546,41 @@ private function check_cached_import( string $path ) : bool {
...
+	}
+
+	/**
+	 * Match each charset on the CSS file.
+	 *
+	 * @param array $match Match array.
+	 *
+	 * @return string
+	 */
+	public function match_charsets( array $match ) : string {
```
Comment (iCaspar) [inline inc/Engine/Optimization/CSSTrait.php]: This is for internal use only, should be `private`

**Review by iCaspar — CHANGES_REQUESTED** [review]

**Review by iCaspar — APPROVED** [review]

**Review by piotrbak — APPROVED** [review]
Good to go! 🎉 
[testrail-report-168.pdf](https://github.com/wp-media/wp-rocket/files/6365426/testrail-report-168.pdf)

**Pull request merged**